common/patches/rof.cpp

```diff
@@ -2651,48 +2653,70 @@ namespace RoF
 			add_member->flags[3] = in_add_member->flags[3];
 			add_member->flags[4] = in_add_member->flags[4];
 			dest->FastQueuePacket(&outapp);
+			safe_delete(outapp);
+			break;
 		}
-		else if (raid_gen->action == 35)
+		case 35:
 		{
-			RaidMOTD_Struct *inmotd = (RaidMOTD_Struct *)__emu_buffer;
-			auto outapp = new EQApplicationPacket(OP_RaidUpdate, sizeof(structs::RaidMOTD_Struct) +
-										 strlen(inmotd->motd) + 1);
-			structs::RaidMOTD_Struct *outmotd = (structs::RaidMOTD_Struct *)outapp->pBuffer;
-
-			outmotd->general.action = inmotd->general.action;
-			strn0cpy(outmotd->general.player_name, inmotd->general.player_name, 64);
-			strn0cpy(outmotd->motd, inmotd->motd, strlen(inmotd->motd) + 1);
+			RaidMOTD_Struct* emu = (RaidMOTD_Struct*)__emu_buffer;
+			auto outapp = new EQApplicationPacket(OP_RaidUpdate, sizeof(structs::RaidMOTD_Struct));
+			structs::RaidMOTD_Struct* eq = (structs::RaidMOTD_Struct*)outapp->pBuffer;
+
+			OUT(general.action);
+			OUT_str(general.player_name);
+			OUT_str(general.leader_name);
+			strncpy(eq->motd, emu->motd.c_str(), emu->motd.length());
 			dest->FastQueuePacket(&outapp);
+			safe_delete(outapp);
+			break;
 		}
-		else if (raid_gen->action == 14 || raid_gen->action == 30)
+		case 14:
+		case 30:
 		{
-			RaidLeadershipUpdate_Struct *inlaa = (RaidLeadershipUpdate_Struct *)__emu_buffer;
-			auto outapp =
-			    new EQApplicationPacket(OP_RaidUpdate, sizeof(structs::RaidLeadershipUpdate_Struct));
-			structs::RaidLeadershipUpdate_Struct *outlaa = (structs::RaidLeadershipUpdate_Struct *)outapp->pBuffer;
+			RaidLeadershipUpdate_Struct* inlaa = (RaidLeadershipUpdate_Struct*)__emu_buffer;
+			auto outapp = new EQApplicationPacket(OP_RaidUpdate, sizeof(structs::RaidLeadershipUpdate_Struct));
+			structs::RaidLeadershipUpdate_Struct* outlaa = (structs::RaidLeadershipUpdate_Struct*)outapp->pBuffer;
 
 			outlaa->action = inlaa->action;
 			strn0cpy(outlaa->player_name, inlaa->player_name, 64);
 			strn0cpy(outlaa->leader_name, inlaa->leader_name, 64);
 			memcpy(&outlaa->raid, &inlaa->raid, sizeof(RaidLeadershipAA_Struct));
 			dest->FastQueuePacket(&outapp);
+			safe_delete(outapp);
+			break;
 		}
-		else
+		case raidSetNote:
+		{
+			auto in_note = (RaidNote_Struct*)__emu_buffer;
+			auto outapp = new EQApplicationPacket(OP_RaidUpdate, sizeof(structs::RaidNote_Struct));
+			auto note = (structs::RaidNote_Struct*)outapp->pBuffer;
+			note->general.action = raidSetNote;
+			strn0cpy(note->general.leader_name, in_note->general.leader_name, sizeof(note->general.leader_name));
+			strn0cpy(note->general.player_name, in_note->general.player_name, sizeof(note->general.leader_name));
+			strncpy(note->note, in_note->note.c_str(), in_note->note.length());
+			dest->QueuePacket(outapp);
+			safe_delete(outapp);
```

Comment:
Can just use `FastQueuePacket` here and remove the `safe_delete` to be a little more efficient. `QueuePacket` will copy the buffer while `FastQueuePacket` will steal it and delete it